runtime/string_functions.cpp

```diff
@@ -1957,15 +1957,29 @@ string f$str_repeat(const string &s, int64_t multiplier) {
   return result;
 }
 
-static string str_replace_char(char c, const string &replace, const string &subject, int64_t &replace_count) {
+static string str_replace_char(char c, const string &replace, const string &subject, int64_t &replace_count, bool with_case) {
   int count = 0;
-  const char *piece = subject.c_str(), *piece_end = subject.c_str() + subject.size();
+  const char *piece = subject.c_str();
+  const char *piece_end = subject.c_str() + subject.size();
+
   string result;
   if (!replace.empty()) {
     result.reserve_at_least(subject.size());
   }
+
   while (true) {
-    const char *pos = (const char *)memchr(piece, c, piece_end - piece);
+    const char *pos = nullptr;
+    if (with_case) {
+      pos = static_cast<const char *>(memchr(piece, c, piece_end - piece));
```

Comment:
Why do we need `memchr` here? Can we change it to `strchr`?

Reply:
It seems we can.

Reply:
Then let's do it like this:
```suggestion
      pos = strchr(piece, c);
```

Reply:
I've thought, that memchr() works a bit faster.

Reply:
> I've thought, that memchr() works a bit faster.

My small benchmarks showed that up to 10,000,000 cycles, 
`strchr` works 62ms against 74ms, but by 100,000,000 cycles 
`strchr` is slower, 1013ms against 932ms for `memchr`.

It seems to me that this is not a big difference and we can 
take the `strchr` function simply because it looks simpler.